### PR TITLE
[webhook/vault-env] Support Vault Transit Secret Engine for PODs

### PR DESCRIPTION
**Pull request by leominov**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| API breaks?     | no
| Deprecations?   | no
| License         | Apache 2.0

### What's in this PR?

Support Transit Secret Engine in environment variables.

P.S.: Looks like whole app must be rewritten.

### Why?

Our developers haven't rich access to Vault, with this changes they can able to encrypt data (we give write access to `encrypt` endpoint of transit engine) and Push it into specs without us. Role used in cluster should have write access to `decrypt` endpoint.

### Checklist

- [x] Error handling code meets the [guideline](https://github.com/banzaicloud/pipeline/blob/master/docs/error-handling-guide.md)

### CODE REVIEW

**Comment by bonifaido** [comment]
Great feature @leominov, I'm okay with it! Could you please provide a working example app that uses this feature? (Also some operator CR which configures the backend accordingly?)

**Comment by leominov** [comment]
Sure

**Comment by leominov** [comment]
@bonifaido I'll also tweak webhook :)

File: cmd/vault-env/main.go [hunk 1]
```diff
@@ -107,6 +111,7 @@ func main() {
 	}
 
 	ignoreMissingSecrets := os.Getenv("VAULT_IGNORE_MISSING_SECRETS") == "true"
+	ignoreTransitDecryptError := os.Getenv("VAULT_IGNORE_TRANSIT_DECRYPT_ERROR") == "true"
```
Comment (bonifaido) [inline cmd/vault-env/main.go]: I'm not sure, maybe we can just reuse `VAULT_IGNORE_MISSING_SECRETS` here?

Comment (leominov) [reply cmd/vault-env/main.go]: Good compromise I think

**Comment by leominov** [comment]
Wow, great!

**Pull request merged**